packages/form-upload/Upload.js

```diff
@@ -58,22 +38,39 @@ const Upload = ({
 
   const fieldValue = Array.isArray(field.value) ? field.value : [];
 
+  const fileDeliveryProps =
+    rest.deliveryChannel && rest.metadata
+      ? {
+          deliveries: [
+            {
+              deliveryChannel: rest.deliveryChannel,
+              metadata: rest.metadata,
+            },
+          ],
+        }
+      : undefined;
+
+  const callFileDelivery = (data, config, upload) => {
+    if (!Array.isArray(upload)) upload = [upload];
+    upload.forEach(upload => {
+      data.deliveries[0].fileURI = upload.references[0];
+      try {
+        avFilesDeliveryApi.uploadFilesDelivery(data, config);
+      } catch (error) {
+        setFieldError(error);
+      }
+    });
+  };
+
   useEffect(() => {
-    if (rest.fileDeliveryOnSubmit && isSubmitting === true) {
-      fileDeliveryOnSubmit(
-        rest.fileDeliveryProps,
+    if (rest.deliverFileOnSubmit && isSubmitting === true) {
+      callFileDelivery(
+        fileDeliveryProps,
         { clientId: rest.clientId, customerId: rest.customerId },
         fieldValue
       );
     }
-  }, [
-    fieldValue,
-    isSubmitting,
-    rest.clientId,
-    rest.customerId,
-    rest.fileDeliveryOnSubmit,
-    rest.fileDeliveryProps,
-  ]);
+  }, [isSubmitting]);
```

Comment:
We want to make sure all the proper useEffect hook dependencies are included here

Reply:
which ones are considered crucial here ? VSCode is recommending all of these. "React Hook useEffect has missing dependencies: 'callFileDelivery', 'fieldValue', 'fileDeliveryProps', 'rest.clientId', 'rest.customerId', and 'rest.deliverFileOnSubmit'. Either include them or remove the dependency array." TBH, I thought that the only one that would actually matter here is the `isSubmitting`